src/serial.h

```diff
@@ -0,0 +1,43 @@
+#ifndef SERIAL_H
+#define SERIAL_H
+
+#include <pthread.h>
+#include <stdint.h>
+
+#define COM1_PORT_BASE 0x03f8
+#define COM1_PORT_SIZE 8
+#define COM1_PORT_END (COM1_PORT_BASE + COM1_PORT_SIZE)
+
+#define FIFO_LEN 64
+
+typedef struct serial_dev serial_dev_t;
+struct serial_dev {
+    uint8_t dll;
```

Comment:
Can you replace these `uint8_t` objects with single `priv`? That is, the member `void *priv` which points to the real storage for serial implementation. Thus, we can encapsulate implementation details in public headers.